### ISSUE TITLE
Allow usage of import "with" attributes

### ISSUE THREAD

**Issue by anon-123456789** [mozilla/addons-linter#5896]
Using [import "with" attibutes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with) causes addons-linter to emit an error like this:
```
Code              Message                                    Description                                                                        File        Line   Column
JS_SYNTAX_ERROR   JavaScript syntax error (Parsing as        There is a JavaScript syntax error in your code, which might be related to some    index.js    1      1
                  module error: Unexpected token with at     experimental JavaScript features that aren't an official part of the language
                  line: 2 and column: 34) (Parsing as        specification and therefore not supported yet. The validation cannot continue on
                  script error: 'import' and 'export' may    this file.
                  appear only with 'sourceType: module' at
                  line: 1 and column: 1)
```
Both static and dynamic imports can cause this issue:
```javascript
import data from "./data.json" with { type: "json" }
```
```javascript
const data = await import("./module.json", { with: { type: "json" } })
```

My addon keeps on getting rejected because of this issue.
